presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFilter.java

```diff
@@ -72,21 +73,21 @@ public LocalDynamicFilter(Multimap<String, Symbol> probeSymbols, Map<String, Int
 
         this.resultFuture = SettableFuture.create();
 
-        this.result = TupleDomain.none();
-        this.partitionsLeft = partitionCount;
+        this.partitionCount = partitionCount;
+        this.partitions = new ArrayList<>(partitionCount);
     }
 
     private synchronized void addPartition(TupleDomain<String> tupleDomain)
     {
         // Called concurrently by each DynamicFilterSourceOperator instance (when collection is over).
-        partitionsLeft -= 1;
-        verify(partitionsLeft >= 0);
+        verify(partitions.size() < partitionCount);
         // NOTE: may result in a bit more relaxed constraint if there are multiple columns and multiple rows.
         // See the comment at TupleDomain::columnWiseUnion() for more details.
-        result = TupleDomain.columnWiseUnion(result, tupleDomain);
-        if (partitionsLeft == 0) {
+        partitions.add(tupleDomain);
+        if (partitions.size() == partitionCount) {
+            Map<Symbol, Domain> result = convertTupleDomain(TupleDomain.columnWiseUnion(partitions));
             // No more partitions are left to be processed.
-            verify(resultFuture.set(convertTupleDomain(result)), "dynamic filter result is provided more than once");
+            verify(resultFuture.set(result), "dynamic filter result is provided more than once");
```

Comment:
This `verify` can never fail as there is no way to enter this `if` block multiple times.

Reply:
Thanks, fixed in d7f7cdd.